### PR TITLE
Change of date dataset structure from dateDataset to dateDateseRef in buckets

### PR DESCRIPTION
**Pull request by patrikbraborec**
JIRA: ONE-4865

---

Supported PR commands:

| Command                | Description            |
| ---------------------- | ---------------------- |
| `ok to test`           | Re-run standard checks |
| `extended test`        | BackstopJS tests       |
| `extended check sonar` | SonarQube tests        |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/2344/ : SUCCESS in 13s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/3966/ : FAILURE in 15m 04s

**Comment by patrikbraborec** [comment]
extended check sonar

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/2379/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/4000/ : FAILURE in 11m 44s

**Comment by patrikbraborec** [comment]
extended check sonar

File: libs/sdk-ui-ext/src/internal/tests/mocks/referencePointMocks.ts [hunk 1]
```diff
@@ -1786,38 +1772,22 @@ export const dateAttributesOnRowsReferencePoint: IReferencePoint = {
...
                     localIdentifier: "date1",
                 },
                 {
                     ...dateItem,
-                    dateDataset: {
-                        ref: {
-                            uri: "closed",
-                        },
-                    },
+                    dateDatasetRef: uriRef("created"),
```
[the remark this answers is not in the document]
Comment (patrikbraborec) [reply libs/sdk-ui-ext/src/internal/tests/mocks/referencePointMocks.ts]: Fixed.

File: libs/sdk-ui-ext/src/internal/tests/mocks/referencePointMocks.ts [hunk 2]
```diff
@@ -1786,38 +1772,22 @@ export const dateAttributesOnRowsReferencePoint: IReferencePoint = {
...
                     localIdentifier: "date2",
                 },
                 {
                     ...dateItem,
-                    dateDataset: {
-                        ref: {
-                            uri: "snapshot",
-                        },
-                    },
+                    dateDatasetRef: uriRef("created"),
```
[the remark this answers is not in the document]
Comment (patrikbraborec) [reply libs/sdk-ui-ext/src/internal/tests/mocks/referencePointMocks.ts]: Fixed.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/2382/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/4003/ : SUCCESS in 13m 18s

**Comment by yenkins** [comment]
### Sonar scan result
 - No new bug found. Congratulation!

More detail, see in https://sonarqube-dev.intgdc.com/dashboard?id=gooddata-ui-sdk-check-extended-sonar-PR1003

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/2383/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-el7/81/ : SUCCESS in 12m 59s (non-voting)

**Comment by patrikbraborec** [comment]
extended test

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/2385/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/659/ : SUCCESS in 21m 27s

**Comment by yenkins** [comment]
### Sonar scan result
 - No new bug found. Congratulation!

More detail, see in https://sonarqube-dev.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR1003

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/725/ : FAILURE in 1m 28s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/988/ : SUCCESS in 16s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/4008/ : SUCCESS in 14m 29s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-el7/83/ : SUCCESS in 8m 33s (non-voting)

**Pull request merged**